src/runmode-dpdk.c

```diff
@@ -1269,13 +1495,27 @@ static void *ParseDpdkConfigAndConfigureDevice(const char *iface)
         FatalError(SC_ERR_DPDK_CONF, "DPDK configuration could not be parsed");
     }
 
-    if (DeviceConfigure(iconf) != 0) {
-        iconf->DerefFunc(iconf);
-        retval = rte_eal_cleanup();
-        if (retval != 0)
-            FatalError(SC_ERR_DPDK_EAL_INIT, "EAL cleanup failed: %s", strerror(-retval));
+    if (iconf->op_mode == DPDK_RING_MODE) {
+        if (DeviceRingsAttach(iconf) != 0) {
+            iconf->DerefFunc(iconf);
+            FatalError(SC_ERR_DPDK_CONF, "Device %s fails to configure", iface);
+        }
+    } else if (iconf->op_mode == DPDK_ETHDEV_MODE) {
+        retval = rte_eth_dev_get_port_by_name(iconf->iface, &iconf->port_id);
+        if (retval < 0) {
+            iconf->DerefFunc(iconf);
+            FatalError(SC_ERR_DPDK_CONF, "Interface \"%s\" unable to obtain port ID: %s",
+                    iconf->iface, rte_strerror(-retval));
+        }
 
-        FatalError(SC_ERR_DPDK_CONF, "Device %s fails to configure", iface);
+        if (DeviceConfigure(iconf) != 0) {
+            iconf->DerefFunc(iconf);
+            retval = rte_eal_cleanup();
+            if (retval != 0)
+                FatalError(SC_ERR_DPDK_EAL_INIT, "EAL cleanup failed: %s", strerror(-retval));
```

Comment:
I guess the error here should say something like "device %s failed to configure, additionally couldn't properly shutdown eal"?